src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -5,10 +5,31 @@
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_PASSWORD_LENGTH = 6;
+    private static final int MIN_AGE = 18;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (user == null) {
+            throw new RuntimeException("User can`t be null");
+        }
+        if (user.getLogin() == null || user.getLogin().isEmpty()) {
```

Comment:
```suggestion
        if (user.getLogin() == null || user.getLogin().isEmpty()) {
```

maybe you will try to add here checking, if login contains/matches empty spaces? because its not good if someone will make his/her login "S    h    i ".(between each letter at least 2-3 spaces, but here can't see, don't know why)

Reply:
I think that login can be "Anna Popova" and in this case empty spaces can be needed

Reply:
Okay, cause i thought login needs to be like login in IG or twitter ahaha no spaces can add:)))